builder/struct.go

```diff
@@ -39,6 +39,7 @@ func (*Struct) Build(gen Generator, ctx *MethodContext, sourceID *xtype.JenID, s
 	usedSourceID := false
 	for i := 0; i < target.StructType.NumFields(); i++ {
```

Comment:
It should be allowed to define matchTag multiple times. E.g. this should work:
```go
// goverter:converter
type Converter interface {
	// goverter:matchTag json
	// goverter:matchTag yaml
	Convert(source Input) Output
}

type Input struct {
	A1 int `json:"a"`
	A2 int `json:"b"`
}

type Output struct {
	B1 int `yaml:"a"`
	B2 int `yaml:"b"`
}
```

Reply:
Or perhaps `goverter:matchTag json yaml` applied in order of precedence?

I did think about having the matchTags be `[]string` vs `string`, but there's a bit of extra complexity here. This could become a bit of a NxM comparison in cases like:
```
type Input struct {
	A1 int `json:"a" yaml:"p"`
	A2 int `json:"b"`
}

type Output struct {
	B1 int `json:"p" yaml:"a"`
	B2 int `yaml:"b"`
}
```

Does `json yaml` or `json\nyaml\n` imply precedence, in which case A1 would fail to match B1, or is the comparison exhaustive? (And would fail in the note you made about ambiguity causing an error.)

The multi-tag use case feels a bit more rare, so I was hesitant to add the complexity. But I'm happy to add it.

Reply:
I agree it's likely rare, but I don't think it adds that much complexity after https://github.com/jmattheis/goverter/pull/164#discussion_r1835364149 is supported.

There shouldn't be precedence, it should error like described here https://github.com/jmattheis/goverter/pull/164#discussion_r1835364149

Basically all/most ambiguous cases are an error in goverter. E.g.
- duplicated enum values https://github.com/jmattheis/goverter/blob/main/scenario/enum_duplicated_different_values.yml
- multiple field name matches
  - https://github.com/jmattheis/goverter/blob/main/scenario/match_ignore_case_unresolved_ambiguity.yml
  - https://github.com/jmattheis/goverter/blob/main/scenario/auto_map_ambigious.yml